### PR TITLE
Parsers-V2: Refactor XFeat.

### PR DESCRIPTION
**Pull request by kkeroo**
This PR refactors XFeat parsers. We introduce `XFeatBaseParser` that inherits from `BaseParser` and also defines some common functions. Additionally, `XFeatBaseParser` defines one additional input because stereo version needs two inputs and instead of changing our `BaseParser` I rather implemented it here. Mono and stereo parser classes are merged into one file along with `XFeatBaseParser` class.

### CODE REVIEW

File: depthai_nodes/ml/parsers/xfeat.py [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+            output_layer_heatmaps,
+            original_size,
+            input_size,
+            max_keypoints,
+        )
+
+        self.previous_results = None
+        self.trigger = False
+
+    def build(
```
Comment (jkbmrz) [inline depthai_nodes/ml/parsers/xfeat.py:300]: is it even necessary to have the `build()` method here if it adds nothing new to the one defined in the parent class?

File: depthai_nodes/ml/parsers/xfeat.py [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+        self.max_keypoints = head_config.get("max_keypoints", self.max_keypoints)
+
+        return self
+
+    def setOutputLayerFeats(self, output_layer_feats):
+        """Sets the output layer containing features.
+
+        @param output_layer_feats: Name of the output layer containing features.
+        @type output_layer_feats: str
+        """
```
Comment (jkbmrz) [inline depthai_nodes/ml/parsers/xfeat.py:131]: should type checking be added to the setters?

File: depthai_nodes/ml/parsers/xfeat.py (hunk 1 above)
Comment (aljazkonec1) [inline depthai_nodes/ml/parsers/xfeat.py:300]: No need to override the inherited build method here as it doesnt change the behavior of build, just calls the super method.

**Review by jkbmrz — APPROVED** [review]
LGTM. Just adding two non-functional comments.

File: depthai_nodes/ml/parsers/xfeat.py [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+        super().__init__(
+            output_layer_feats,
+            output_layer_keypoints,
+            output_layer_heatmaps,
+            original_size,
+            input_size,
+            max_keypoints,
+        )
+
+    def build(
```
Comment (aljazkonec1) [inline depthai_nodes/ml/parsers/xfeat.py:445]: same as for mono, no need

**Review by aljazkonec1 — APPROVED** [review]

File: depthai_nodes/ml/parsers/xfeat.py (hunk 3 above)
Comment (kkeroo) [reply depthai_nodes/ml/parsers/xfeat.py]: Added in [55095b3](https://github.com/luxonis/depthai-nodes/pull/104/commits/55095b38d4b1c80694093331dcb4c303ea22a893).

File: depthai_nodes/ml/parsers/xfeat.py (hunk 1 above)
Comment (kkeroo) [reply depthai_nodes/ml/parsers/xfeat.py]: Removed in [55095b3](https://github.com/luxonis/depthai-nodes/pull/104/commits/55095b38d4b1c80694093331dcb4c303ea22a893).

Comment (kkeroo) [reply depthai_nodes/ml/parsers/xfeat.py]: Removed in [55095b3](https://github.com/luxonis/depthai-nodes/pull/104/commits/55095b38d4b1c80694093331dcb4c303ea22a893).

File: depthai_nodes/ml/parsers/xfeat.py (hunk 2 above)
Comment (kkeroo) [reply depthai_nodes/ml/parsers/xfeat.py:131]: Added in [a87701f](https://github.com/luxonis/depthai-nodes/pull/104/commits/a87701f03b1f6b6e83ec4e3c4b12dd124e3cb1cf). Note that we should do this for all parsers.

**Pull request merged**